onnxruntime/core/optimizer/free_dim_override_transformer.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/logging/logging.h"
+#include "core/session/inference_session.h"
+#include "core/graph/graph_utils.h"
+#include "core/optimizer/free_dim_override_transformer.h"
+
+using namespace ONNX_NAMESPACE;
+using namespace ::onnxruntime::common;
+namespace onnxruntime {
+
+static std::string ToLower(std::string s) {
```

Comment:
const std::string& s

Reply:
In this case, the function actually does want to make a copy (since it returns a new string, rather than mutating the original one).

Reply:
I think Changming meant the input parameter should be const reference. Don't need a copy of that.

---
In reply to: [325949343](https://github.com/microsoft/onnxruntime/pull/1825#discussion_r325949343) [](ancestors = 325949343)

Reply:
Either works, doesn't matter